Need an option to build OS64 libraries without code signing

**Describe the bug**
If you are trying to build OS64 or any platform which is not a simulator, you will be forced to code sign the output. When building from a Mac without code signing certificate (no Apple developer account), the CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM will be set as "123456789A" and the build will fail with error message,

Code Signing Error: No signing certificate "iOS Development" found:  No "iOS Development" signing certificate matching team ID "123456789A" with a private key was found.

**To Reproduce**
Steps to reproduce the behavior:
1. Build any project for device (not simulator) such as OS64 without providing CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM 

**Expected behavior**
Be able build without code signing.
Not every libraries are required to be code signed, for example,
- Static linked libraries do not require code signing
- App developers may choose to code sign the dynamic library manually late
- For CIs verifying building, adding code signing will be extra work without obvious benefits

**Environment**
* What version of CMake did you run?
* Your faulty CMakeLists.txt here:
```cmake
--- Paste here ---
```
**Proposal**
Add an extra option to disable code signing, if CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM is not set.
Can send PR

I tried this fix. I just get a different error. I understand that I am selecting "Xcode" as my generator...should I do something different for static libraries. Thanks!
```
Check dependencies
Code Signing Error: Signing for "argonishche_sh" requires a development team. Select a development team in the Signing & Capabilities editor.

** BUILD FAILED **
```

I am trying to compile an open source library argon2. I will be linking it with other C code and eventually will be calling the code from iOS. I've done this before on Android but no experience on iOS. Here is the simple CMakeLists.txt file I'm testing on:
```cmake
cmake_minimum_required(VERSION 3.1)
  
project(external-argon2)
include(ExternalProject)

list(APPEND CMAKE_ARGS
    "-DCMAKE_TOOLCHAIN_FILE:PATH=${CMAKE_TOOLCHAIN_FILE}"
)

if(IOS)
    list(APPEND CMAKE_ARGS
        "-DPLATFORM=${PLATFORM}"
    )
endif()

ExternalProject_add(
    argon2
    URL https://github.com/yandex/argon2/archive/master.zip
    PREFIX argon2
    CMAKE_ARGS "${CMAKE_ARGS}"
    )

```

I execute the following cmake command taken from your examples
`cmake .. -G Xcode -DCMAKE_TOOLCHAIN_FILE=/Volumes/D1/dev/proj/.../cmake-ios/ios.toolchain.cmake -DPLATFORM=OS64`

--- a/ios.toolchain.cmake
+++ b/ios.toolchain.cmake
@@ -365,9 +365,6 @@ endif()
 # Set Xcode property for SDKROOT as well if Xcode generator is used
 if(USED_CMAKE_GENERATOR MATCHES "Xcode")
   set(CMAKE_OSX_SYSROOT "${SDK_NAME}" CACHE INTERNAL "")
-  if(NOT DEFINED CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM)
-    set(CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM "123456789A")
-  endif()
 endif()
 
 # Specify minimum version of deployment target.

